pkg/controller/atlasproject/project.go

```diff
@@ -16,26 +15,50 @@ func ensureProjectExists(ctx *workflow.Context, connection atlas.Connection, pro
 	if err != nil {
 		return "", workflow.Terminate(workflow.Internal, err.Error())
 	}
+	// Try to find the project
+	projectID, err := findProject(connection, project, client)
```

Comment:
Please consider using [mongodbatlas.GetOneProjectByName()](https://github.com/mongodb/go-client-mongodb-atlas/blob/85a3c4de47becac8fa7887bea382190833c4ee0f/mongodbatlas/projects.go#L135) instead

Reply:
This is a very bad endpoint and we had issues with it in Enterprise Operator. I don't remember details though
The good alternative is to use the "getProjects(name=<foo>)" but this is not supported by atlas client (though is exposed by the API)

Reply:
After discussing with @gssbzn it seems like this endpoint should work ok for organization programmatic API keys - will adress